cmd/thanos/main.go

```diff
@@ -214,3 +210,22 @@ func defaultGRPCServerOpts(logger log.Logger, reg *prometheus.Registry, tracer o
 		),
 	}
 }
+
+func metricHTTPListenGroup(g *run.Group, logger log.Logger, reg *prometheus.Registry, httpBindAddr string) error {
```

Comment:
I'd add a comment to this function that it's called from `compact.go`

Reply:
its run from a few places ... personally think i would just add this to the run group in main by doing something like each `registerXXX` returning a func to be added then finally adding this metrics as I think it is probably wanted for all components.

Reply:
We cannot add it everywhere, since not everyhing exposes HTTP Port as well not everything has JUST metric in the route mux. Will add comment.